base/abstractset.jl

```diff
@@ -287,6 +281,13 @@ function symdiff!(s::AbstractSet, itrs...)
 end
 
 function symdiff!(s::AbstractSet, itr)
+    for x in unique(itr)
```

Comment:
won't this allocate a bunch and be slow?

Reply:
Yeah, but for the general case, hashmap-based solutions always other methods in my tests, and we need to put the hashmap somewhere.

We can also do 

```
symdiff!(s::AbstractSet, itr) = symdiff!(s::AbstractSet, Set(itr))
```

I think that's the same though, since `unique` just makes a `Set` under the hood. I guess `unique` would allocate an additional vector though, you're right. 

For types that can be radix-sorted, I have the following method:

```
function symdiff!(A, _B)
    issorted(A) || sort!(A)
    B = issorted(_B) ? _B : sort(_B)
    N = length(A)
    M = length(B)
    i = 1
    j = 1
    A_out_ind = 1
    B_out_ind = 1
    @inbounds while i <= N && j <= M
        if A[i] < B[j]
            A[A_out_ind] = A[i]
            A_out_ind += 1
            i = _lookahead(A[i], A, i)
        elseif B[j] < A[i]
            B[B_out_ind] = B[j]
            B_out_ind += 1
            j = _lookahead(B[j], B, j)
        else
            i = _lookahead(A[i], A, i)
            j = _lookahead(B[j], B, j)
        end
    end
    @inbounds while i <= N
        A[A_out_ind] = A[i]
        A_out_ind += 1
        i = _lookahead(A[i], A, i)
    end
    @inbounds while j <= M
        B[B_out_ind] = B[j]
        B_out_ind += 1
        j = _lookahead(B[j], B, j)
    end
    resize!(A, A_out_ind + B_out_ind - 2)
    copyto!(A, A_out_ind, B, 1, B_out_ind - 1)
end

function _lookahead(cur, l, ind)
    @inbounds while ind <= length(l)
        if l[ind] > cur
            return ind
        else
            ind += 1
        end
    end
    return ind
end
```

but this idea can be applied to *all* the Set operations that take `Vector`, which is the subject of another issue.

Given that `symdiff` is currently broken, I think that it's best to get correct behauviour first, and then